### PR TITLE
[EASI-3610] Update gainshare eligibility question

### PR DESCRIPTION
**Pull request by OddTomBrooks**
# EASI-3610

## Changes and Description

Fixes bug where export modal doesnt open on the view view

Backend Changes:

- Updated gql schema
- Created flyway migration
- Updated data model
- Updated unit test
- Updated postman collection

Frontend Changes:

- Updated GQL queries to include new fields 
  - `gainsharePaymentsEligibility`
  - `gainsharePaymentsEligibilityOther`
- Updated forms to include new condition question
- Updated readonly to render new data
- Updated CSV to include new fields
- Updated unit tests
- Updated cypress tests

## How to test this change

Backend Instructions:

- Start containers
- Verify plan participants and providers have new columns gainsharePaymentsEligibility and gainsharePaymentsEligibilityOther
- Run Postman
- Verify you can set appropriate values for the corresponding columns

Frontend Instructions:

- Log in and navigate to page 4 (Coordination) of Participants and Providers
- Click `Yes` to gainsharing payments question
- Verify new question is listed as a conditional followup - `Are providers eligible to participate in gainsharing arrangements?`
- Verify if selecting `Other` that the conditional text field appears
- Verify selections persist and render on the form upon reentry
- Verify data is rendered in readonly view
- Verify data is rendered in CSV export

## PR Author Review Checklist

- [x] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [x] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [x] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

**Review by StevenWadeOddball — CHANGES_REQUESTED** [review]
@OddTomBrooks, this looks good, only the migration data type should be updated and then I think the BE is ready!

@patrickseguraoddball , I don't see the new question in the read-only view. I confirmed it persists if I go into the editable view though. Would you expect to see the question here?  (I haven't tested the CSV export, it appears to be my own local configuration issue)

image

**Comment by patrickseguraoddball** [comment]
> @OddTomBrooks, this looks good, only the migration data type should be updated and then I think the BE is ready!
> 
> @patrickseguraoddball , I don't see the new question in the read-only view. I confirmed it persists if I go into the editable view though. Would you expect to see the question here? (I haven't tested the CSV export, it appears to be my own local configuration issue)
> 

@StevenWadeOddball 
Ah, the readonly work is complete, it is just keying off of the tracking question instead of the main gainshairing payment question.  If you had selected yes for the tracking question it would show in the readonly.  I'll update this to be conditional on the first question, rather than the second

**Comment by patrickseguraoddball** [comment]
https://github.com/CMSgov/mint-app/blob/5c20a221fe6f3c9e6926a770c8c03b5a8ea0469c/src/views/ModelPlan/ReadOnly/ParticipantsAndProviders/index.tsx#L389-L392

Actually had it conditional on both

**Comment by StevenWadeOddball** [comment]
Thanks @patrickseguraoddball! @OddTomBrooks , can you please let me know when you update the migration? I'll validate all changes at the same time.

File: migrations/V119__Add_Gainshare_Eligibility.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+CREATE TYPE GAINSHARE_ARRANGEMENT_ELIGIBILITY AS ENUM (
+  'ALL_PROVIDERS',
+  'SOME_PROVIDERS',
+  'OTHER',
+  'NO'
+  );
+
+ALTER TABLE plan_participants_and_providers
+  ADD COLUMN gainshare_payments_eligibility GAINSHARE_ARRANGEMENT_ELIGIBILITY[] DEFAULT NULL,
+  ADD COLUMN gainshare_payments_eligibility_other TEXT DEFAULT NULL;
```
[the remark this answers is not in the document]
Comment (OddTomBrooks) [reply migrations/V119__Add_Gainshare_Eligibility.sql:10]: Ahaa, thanks for the feedback. Updated on the other branch, but forgot to do the same here. Good catch!

**Review by StevenWadeOddball — DISMISSED** [review]
Thanks @OddTomBrooks and @patrickseguraoddball . 

I tested the read only view and the migration looks great! I did not test the csv export explicitly, but this all looks good and functions as expected!
image

**Review by StevenWadeOddball — APPROVED** [review]

**Pull request merged**